### PR TITLE
[RW-7338] workspace admin UI test case

### PR DESCRIPTION
**Pull request by chavanr2019**
Workspace admin test:

1. Verify  workspace admin ui
2. Verify that admin is able to open an notebookpreview
3. Verify that admin is able to delete runtime

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: e2e/app/page/admin-workspace-page.ts
Comment (chavanr2019) [reply e2e/app/page/admin-workspace-page.ts:75]: This test is only to verify if the workspace-admin UI renders correctly. I'll look into other UI-check test cases and improvise this test.

File: e2e/tests/nightly/admin/workspace-admin-ui.spec.ts
Comment (chavanr2019) [reply e2e/tests/nightly/admin/workspace-admin-ui.spec.ts:68]: This is a notebook python file. I'll add a comment about it.

File: e2e/tests/nightly/admin/workspace-admin-ui.spec.ts
Comment (aweng98) [inline e2e/tests/nightly/admin/workspace-admin-ui.spec.ts:76]: This test coverage is no longer testing `workspace-admin-ui` page. It should be removed or belongs in a different test suite.

File: e2e/tests/nightly/admin/workspace-admin-ui.spec.ts
Comment (aweng98) [inline e2e/tests/nightly/admin/workspace-admin-ui.spec.ts:80]: Missing `await` here.

File: e2e/tests/nightly/admin/workspace-admin-ui.spec.ts
Comment (aweng98) [inline e2e/tests/nightly/admin/workspace-admin-ui.spec.ts:96]: Missing `await`.

File: e2e/app/page/admin-notebook-preview-page.ts
Comment (chavanr2019) [reply e2e/app/page/admin-notebook-preview-page.ts:9]: ok! will check. Thanks!

**Review by chavanr2019 — COMMENTED** [review]
Hi Joel & Alex, I'll update the test case, thanks for reviewing!

File: e2e/app/modal/lock-workspace-modal.ts
Comment (chavanr2019) [inline e2e/app/modal/lock-workspace-modal.ts:8]: New "modal" file to cover the LOCK WORKSPACE feature

File: e2e/app/page/admin-workspace-page.ts
Comment (chavanr2019) [inline e2e/app/page/admin-workspace-page.ts]: Deleted these function since they were only verifying the Headings of the workspace-admin page.

File: e2e/app/page/admin-workspace-page.ts
Comment (chavanr2019) [inline e2e/app/page/admin-workspace-page.ts]: Deleted these function since they were only verifying the Headings of the workspace-admin page.

File: e2e/app/page/admin-notebook-preview-page.ts
Comment (chavanr2019) [reply e2e/app/page/admin-notebook-preview-page.ts:9]: @aweng98 in [slack chat](https://pmi-engteam.slack.com/archives/C0287U4HEQ4/p1634918962001500)  you'd asked me to use the same pattern as [here](https://github.com/all-of-us/workbench/blob/4d4cddbc70fbc82c7ec4808d472bb7bb23a63354/e2e/utils/test-utils.ts#L39): but seems like it is beyond my coding knowledge

File: e2e/app/page/admin-workspace-page.ts
Comment (chavanr2019) [reply e2e/app/page/admin-workspace-page.ts]: will fix this.

File: e2e/tests/nightly/admin/workspace-admin-ui.spec.ts
Comment (chavanr2019) [inline e2e/tests/nightly/admin/workspace-admin-ui.spec.ts:126]: The test is failing here, since it ends before the status changes from "running" to "deleting"

**Review by chavanr2019 — COMMENTED** [review]
@aweng98 @jmthibault79 Please review whenever time permits. Thank!

File: e2e/app/page/admin-workspace-page.ts
Comment (jmthibault79) [inline e2e/app/page/admin-workspace-page.ts:69]: this is not used

File: e2e/app/page/admin-workspace-page.ts
Comment (jmthibault79) [inline e2e/app/page/admin-workspace-page.ts:109]: what does this return when there are active runtimes?  Does it make sense to add a check in here for the `noActiveRuntimeText` and return a `Promise<boolean>` instead?

File: e2e/app/page/admin-notebook-preview-page.ts
Comment (jmthibault79) [reply e2e/app/page/admin-notebook-preview-page.ts:9]: We could save that for a follow-up PR.  Alex, what do you think?

File: e2e/app/page/admin-workspace-page.ts
Comment (chavanr2019) [reply e2e/app/page/admin-workspace-page.ts:69]: agree! will remove this. Thanks!

File: e2e/tests/nightly/admin/workspace-admin-ui.spec.ts
Comment (chavanr2019) [reply e2e/tests/nightly/admin/workspace-admin-ui.spec.ts:129]: will update!

File: e2e/app/modal/lock-workspace-modal.ts
Comment (chavanr2019) [reply e2e/app/modal/lock-workspace-modal.ts:43]: agree!

File: e2e/app/page/admin-workspace-page.ts
Comment (chavanr2019) [reply e2e/app/page/admin-workspace-page.ts:109]: makes sense. Let me try that!

File: e2e/app/modal/delete-runtime.modal.ts
Comment (chavanr2019) [reply e2e/app/modal/delete-runtime.modal.ts:8]: @aweng98  I think now I get what you meant in your above comment. The deleteRuntime in the runtime-panel.ts is different compared this one. In the runtime-panel there is NO modal that pops on clicking deleteEnvironment.

File: e2e/app/modal/lock-workspace-modal.ts
Comment (chavanr2019) [reply e2e/app/modal/lock-workspace-modal.ts:43]: agree!

File: e2e/app/page/admin-workspace-page.ts
Comment (chavanr2019) [reply e2e/app/page/admin-workspace-page.ts:69]: agree! will remove this. Thanks!

File: e2e/app/page/admin-workspace-page.ts
Comment (chavanr2019) [reply e2e/app/page/admin-workspace-page.ts:109]: makes sense. Let me try that!

File: e2e/tests/nightly/admin/workspace-admin-ui.spec.ts
Comment (chavanr2019) [reply e2e/tests/nightly/admin/workspace-admin-ui.spec.ts:126]: right, it times out. I can try increasing the timeout.

File: e2e/tests/nightly/admin/workspace-admin-ui.spec.ts
Comment (chavanr2019) [reply e2e/tests/nightly/admin/workspace-admin-ui.spec.ts:129]: will update!

**Review by chavanr2019 — COMMENTED** [review]
@jmthibault79  I'll update the PR as per your review comment. Thanks!

File: e2e/app/page/admin-workspace-page.ts
Comment (chavanr2019) [reply e2e/app/page/admin-workspace-page.ts:102]: @jmthibault79 this function is only checking for the "no active run time running" text before creating the runtime and when there are active runtime - getRuntimeStatus() checks the runtime status. 
Since for every test run we are creating a new workspace and there is no ACTIVE runtime, I was thinking we could remove this function and the verification STEP. Thoughts!

Comment (jmthibault79) [reply e2e/app/page/admin-workspace-page.ts:102]: that seems OK to me

File: e2e/tests/nightly/admin/workspace-admin-ui.spec.ts
Comment (chavanr2019) [reply e2e/tests/nightly/admin/workspace-admin-ui.spec.ts:134]: @jmthibault79 after increasing the timeout all the 3- test cases have passed.

**Review by chavanr2019 — COMMENTED** [review]
the 3-test cases have passed locally. Please review. Thanks!

**Review by jmthibault79 — APPROVED** [review]

File: e2e/app/component/navigation.ts
Comment (chavanr2019) [reply e2e/app/component/navigation.ts:40]: @aweng98  I changed this xpath because 
`//span[contains(., "")]`selects the first workspaces option **YOUR WORKSPACES** , whereas `//span[(text()="${destinationApp}")] `selects the exact match.

**Review by chavanr2019 — COMMENTED** [review]
Hi @aweng98 This test has passed locally and Joel has approved of all the changes. When you've the time, I would like you please review the code / comments where I've tagged you. After you approve I'll merge the PR. Thanks!

File: e2e/app/modal/delete-runtime.modal.ts
Comment (aweng98) [inline e2e/app/modal/delete-runtime.modal.ts:18]: Is `Delete` button enabled when modal is open? If it is, we should have `.waitUntilEnabled()` in `isLoaded`. And `await waitWhileLoading(this.page);`.

File: e2e/app/modal/lock-workspace-modal.ts
Comment (aweng98) [inline e2e/app/modal/lock-workspace-modal.ts:20]: Same comment from above for this `isLoaded`.

File: e2e/app/modal/lock-workspace-modal.ts
Comment (aweng98) [inline e2e/app/modal/lock-workspace-modal.ts:25]: Wrong func name.

File: e2e/app/page/admin-notebook-preview-page.ts
Comment (aweng98) [reply e2e/app/page/admin-notebook-preview-page.ts:9]: Yes, a follow-up PR is a good idea.

File: e2e/app/page/admin-notebook-preview-page.ts
Comment (aweng98) [inline e2e/app/page/admin-notebook-preview-page.ts:17]: As a good practice, calling `waitWhileLoading` at the end of `isLoaded` func.

File: e2e/app/page/workspace-data-page.ts
Comment (aweng98) [inline e2e/app/page/workspace-data-page.ts:227]: `page` is a global object. It should be `this.page`. Please check and make sure no other pages are using global `page` object.
`console.log` is not needed.

File: e2e/tests/nightly/admin/workspace-admin-ui.spec.ts
Comment (aweng98) [inline e2e/tests/nightly/admin/workspace-admin-ui.spec.ts:29]: Missing `await dataPage.waitForLoad`. Please check other places where this call is also needed.

**Review by aweng98 — APPROVED** [review]
LG! Few nits. Approved assuming test runs and pass.

**Review by chavanr2019 — COMMENTED** [review]
Merging the workspace-admin PR. Thanks Alex & Joel!

**Pull request merged**
